tests/migration_test.go

```diff
@@ -316,6 +316,58 @@ var _ = Describe("Migrations", func() {
 				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
 			})
 		})
+		Context("with an Alpine DataVolume", func() {
+			BeforeEach(func() {
+				tests.BeforeTestCleanup()
+				if !tests.HasCDI() {
+					Skip("Skip DataVolume tests when CDI is not present")
+				}
+			}, 60)
+			It("should reject a migration of a vmi with a non-shared data volume", func() {
```

Comment:
It would be nice to have a functional test that verifies we can migrate with shared datavolumes as well. 

I wonder if we can use the same technique we're using with the iscsi PVs, but have the datavolume populate the iscsi device with an image from that test http server we have that serves up alpine/cirros.

Reply:
I have this test written, but took it out on purpose because I couldn't get it to work.
When I provide an ISCSI PV from our test server it doesn't get populated and the vm fails to start.
So I'm trying to figure out how datavolume populates it.

I can post the test with all the changes I've done to support this - perhaps someone with a better DV knowledge can point out what's wrong before I do...

Reply:
The datavolume http population flow is just creating a pod with the PVC attached to it. That pod then pulls down the image from the http server and stores it on the pvc.

You could test this by attaching the PVC to the pod and attempting to write a file to the PVC.